File: 1291-sequential-digits/1291-sequential-digits.go
func sequentialDigits(low int, high int) []int {
    startDigit := toDigit(low)
    endDigit := toDigit(high)
    results := make([]int, 0)
    
    for i := startDigit; i <= endDigit; i++ {
        for j := 1; j < 9; j++ {
            if i + j > 10 {
                continue
            }
            
            value := toSequential(i, j)
            if low > value || value > high {
                continue
            }
            
            results = append(results, value)
        }
    }
    
    return results;
}

func toDigit(value int) int {
    result := 1
    for value > 9 {
        value = value / 10
        result++
    }
    
    return result
}

func toSequential(size int, startValue int) int {
    result := startValue
    for size > 1 {
        startValue++
        result = result * 10 + startValue
        size--
    }
    
    return result
}

File: 1291-sequential-digits/1291-sequential-digits.java
class Solution {
    public List<Integer> sequentialDigits(int low, int high) {
        int startDigit = toDigit(low);
        int endDigit = toDigit(high);
        List<Integer> results = new ArrayList<>();

        for (int i = startDigit; i <= endDigit; i++) {
            for (int j = 1; j < 9; j++) {
                if (i + j > 10) {
                    continue;
                }

                int value = toSequential(i, j);
                if (value < low || value > high) {
                    continue;
                }
                results.add(value);
            }
        }

        return results;
    }
    
    private int toDigit(int value) {
        int result = 1;
        while (value > 9) {
            value = value / 10;
            result++;
        }
        
        return result;
    }
    
    private int toSequential(int size, int startValue) {
        int result = startValue;
        while (size > 1) {
            startValue++;
            result = result * 10 + startValue;
            size--;
        }
        
        return result;
    }
}